lib/SugarClient/client.js

```diff
@@ -0,0 +1,143 @@
+class SugarClient {
+  static initClass() {
+    this.host = null;
+    this.Primus = null;
+  }
+
+  constructor(userId, authToken) {
+    this.primusUrl = this.primusUrl.bind(this);
+    this.connect = this.connect.bind(this);
+    this.disconnect = this.disconnect.bind(this);
+    this.receiveData = this.receiveData.bind(this);
+    this.subscribeTo = this.subscribeTo.bind(this);
+    this.unsubscribeFrom = this.unsubscribeFrom.bind(this);
+    this.on = this.on.bind(this);
+    this.off = this.off.bind(this);
+    this.emit = this.emit.bind(this);
+    this.__subscribeToChannels = this.__subscribeToChannels.bind(this);
+    this.__subscribeTo = this.__subscribeTo.bind(this);
+    this.createEvent = this.createEvent.bind(this);
+    this.userId = userId;
+    this.authToken = authToken;
+    this.events = { };
+    this.subscriptions = { };
+    this.initializePrimus();
+  }
+
+  initializePrimus() {
+    if (SugarClient.Primus == null) { throw 'SugarClient.Primus is not defined'; }
+    if (SugarClient.host == null) { throw 'SugarClient.host is not defined'; }
+    this.primus = SugarClient.Primus.connect(SugarClient.host, {
+      websockets: true,
+      network: true,
+      manual: true
```

Comment:
The only difference between this and https://github.com/zooniverse/Sugar-Client/blob/master/client.js that I'm seeing is the deletion of `ping: 10000` ([link](https://github.com/zooniverse/Sugar-Client/blob/master/client.js#L44)). Apologies if it has already been stated somewhere, but is that property no longer needed with Primus 8, and it was with Primus 6?

Reply:
Possibly related - https://github.com/primus/primus#protocol

Reply:
It’s been removed because the direction of the ping/pong protocol changed from client->server to server->client. There’s [a list of client options](https://github.com/primus/primus#connecting-from-the-browser) in the Readme.

Reply:
So that option said: ping the server then wait 10s for a ping response. The client now sends pong back in response to pings from the server. 

You can see the traffic if you watch the web socket in browser dev tools.